### PR TITLE
ADL_score_5 derived variable

### PR DESCRIPTION
**Pull request by yulric**
This PR adds the derived variable `ADL_score_5` according to this issue https://github.com/Big-Life-Lab/cchsflow/issues/73.

This variable has 6 categories:
0 - Needs help with 0 tasks
1 - Needs help with at least 1 task
2 - Needs help with at least 2 tasks
3 - needs help with at least 3 tasks
4 - Needs help with at least 4 tasks
5 - Needs help with at least 5 tasks

It uses the CCHS ADL variables that are common accros the cycles from 2001 to 2014 which are ADL_01, ADL_02, ADL_03, ADL_04 and ADL_05

### CODE REVIEW

File: R/adl-score-5.R [hunk 1]
```diff
@@ -0,0 +1,152 @@
+#' @title Derived variable. Number of tasks the individual needs help with
+#'
+#' @description A 6 category variables representing the number of tasks
+#' an individual needs help with. It also has two categories to represent
+#' "missing" and "not applicable". It uses the following five ADL variables
+#' which are common accros CCHS cycles from 2201 to 2014:
```
Comment (wyusuf068) [inline R/adl-score-5.R:6]: From 2001 to 2014.

File: R/adl-score-5.R [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+adl_score_5_fun <-
+  function(ADL_01, ADL_02, ADL_03, ADL_04, ADL_05) {
+    # For each row in the ADL variables, count the total number of "not applicable"
+    # values in each row
+    count_missing_adl <- rep(0, length(ADL_01))
+    # For each row of each of the above ADL variables, if they have a value of 
+    # "not applicable" then increment the count for that row in the above
+    # vector by one
+    count_missing_adl <-
+      ifelse(ADL_01 == "NA::b", count_missing_adl + 1, count_missing_adl)
```
Comment (wyusuf068) [inline R/adl-score-5.R:87]: NA::b is missing. Did you mean to say NA::a to check for "not applicable"?

File: R/adl-score-5.R [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+      ifelse(ADL_02 == "NA::b", count_missing_adl + 1, count_missing_adl)
+    count_missing_adl <-
+      ifelse(ADL_03 == "NA::b", count_missing_adl + 1, count_missing_adl)
+    count_missing_adl <-
+      ifelse(ADL_04 == "NA::b", count_missing_adl + 1, count_missing_adl)
+    count_missing_adl <-
+      ifelse(ADL_05 == "NA::b", count_missing_adl + 1, count_missing_adl)
+    
+    # For each row in the ADL variables, count the total number of missing
+    # values in each row
```
Comment (wyusuf068) [inline R/adl-score-5.R:98]: Count the total number of not applicable values in each row.

**Review by wyusuf068 — APPROVED** [review]
Looks good to me. A few minor documentation fixes before merging to `dev`.

**Review by CBjerke — APPROVED** [review]
Agree with Warsame's comments re edits

**Comment by yulric** [comment]
@wyusuf068 I've made the changes you proposed. Let me know if it all looks good.

**Review by DougManuel — CHANGES_REQUESTED** [review]
No clear best approach, but I suggest that we have a NA category if there is NA for any component. See the screenshot for SCHDSTG below. 

In practice, I would impute missing for the individual questions before calculating this measure, but the imputation practice is not part of cchsflow and there is no definitive approach to address missing data. Allowing a flexible approach to missing data is important and the current or the suggested revision are both flexible.

@CBjerke I reviewed how Statscan considers missing data in derived variables such as `PACDEE` (METS of physical activity) and `SCHDSTG` (Smoking - Stages of change). I'm not sure there is a consistent approach or at least the approach to consider missing data depends on the context of the question. More broadly, there is not a consistent approach that people use for any variable in CCHS. e.g. do you include or exclude NA when estimating the prevalence of smoking? 

For PACDEE, if you assigned a value of zero METS for a specific activity if you are a non-respondent to a specific activity (e.g. gardening). 

For SCHDSTG, if you are assigned NA to the overall score if you respond to NA for any component. 

Screen Shot 2020-10-28 at 1 33 14 PM

File: R/adl-score-5.R [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+#' @return A derived variable (ADL_score_5) with 6 categories:
+#' \enumerate{
+#'   \item 0 - Needs help with 0 tasks
+#'   \item 1 - Needs help with at least 1 task
+#'   \item 2 - Needs help with at least 2 tasks
+#'   \item 3 - Needs help with at least 3 tasks
+#'   \item 4 - Needs help with at least 4 tasks
+#'   \item 5 - Needs help with at least 5 tasks
+#' }
+#' A value of NA::a means that all the input variables had NA::a
```
[the remark this answers is not in the document]
Comment (yulric) [reply R/adl-score-5.R:30]: @DougManuel Not sure what you mean? I though NA::c meant not asked in the survey, NA::a is not applicable.

File: R/adl-score-5.R [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+#'
+#' combined_adl <- bind_rows(adl2001, adl2009_2010)
+#'
+#' head(combined_adl)
+#'
+#' tail(combined_adl)
+#'
+#' # Using adl_score_5_fun() to generate to ADL_score_5 based on user inputted values.
+#' #
+#' # Let's say you do not need help preparing meals, you need help getting to
```
[the remark this answers is not in the document]
Comment (yulric) [reply R/adl-score-5.R:69]: I copied this from Warsame's example for ADL_der and just changed it for this score haha.

**Comment by yulric** [comment]
@DougManuel Should the priority be to set "NA::b" or "NA::a"? i.e. if a individual has missing data for ADL_01 and "not applicable" data for ADL_02 what should the variable be set to?

**Comment by DougManuel** [comment]
> @DougManuel Should the priority be to set "NA::b" or "NA::a"? i.e. if a individual has missing data for ADL_01 and "not applicable" data for ADL_02 what should the variable be set to?

Good question. Suggest NA::a 

I thought that we were transforming from the original cchs variables which have NA values of 6, 7, 8, etc. 

Suggest eight categories in total:

0 - Needs help with 0 tasks
1 - Needs help with at least 1 task
2 - Needs help with at least 2 tasks
3 - needs help with at least 3 tasks
4 - Needs help with at least 4 tasks
5 - Needs help with at least 5 tasks
NA::a - one or more ADL intermediate variables with NA::a
NA::b - one or more ADL intermediate variables with NA::b AND no variables with NA::a

File: R/adl-score-5.R (hunk 4 above)
[the remark this answers is not in the document]
Comment (wyusuf068) [reply R/adl-score-5.R]: Perhaps write: "#' A value of NA::a means that all the input variables had NA::a, not applicable."?

File: R/adl-score-5.R [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+#' # doing personal care, and do not need help moving inside the house. Using
+#' # adl_score_5_fun() we can check the number of tasks you need help with
+#'
+#' ADL_score_5 <- adl_score_5_fun(2, 1, 1, 2, 2)
+#'
+#' print(ADL_score_5)
+#'
+adl_score_5_fun <-
+  function(ADL_01, ADL_02, ADL_03, ADL_04, ADL_05) {
+    # For each row in the ADL variables, count the total number of "not applicable"
```
Comment (wyusuf068) [inline R/adl-score-5.R:81]: count the total number of "missing" values in each row*

**Comment by yulric** [comment]
@DougManuel @wyusuf068 Pushed your changes.

**Review by DougManuel — APPROVED** [review]
Looks very good.
- I made a few edit suggestions for the document. Some of the edit suggestions are to align with the documentation for other functions.

- There is [switch](https://www.rdocumentation.org/packages/base/versions/3.6.2/topics/switch) in base R and case_when in dyplr (a package dependency) that may work if you don't want the if-else statements.

**Pull request merged**